dask/array/tests/test_routines.py

```diff
@@ -1286,3 +1286,54 @@ def test_result_type():
     # dask 0d-arrays are NOT treated like scalars
     c = da.from_array(np.ones((), np.float64), chunks=())
     assert da.result_type(a, c) == np.float64
+
+
+def test_einsum():
```

Comment:
Einsum seems tricky enough that I suggest we consider building an additional test that is parametrized over many different kinds of situations:

Perhaps something like the following (untested):
```python
@pytest.mark.parametrize('signature', [
    'abc,bad->abcd', 
    'ab,bc-> ...' 
    ...
])
def test_einsum(signature):
    dimensions = {'a': 5, 'b': 6, 'c': 7, 'd': 5, 'e': 6}

    def shape_from_string(s):
        return tuple(dimensions[c] for c in s)

    np_inputs = [np.random.random(shape_from_string(s) for s in signature.split('->')[0].split(',')]
    da_inputs = [da.from_array(x, chunks=4) for x in np_inputs]

    assert_eq(np.einsum(signature, *np_inputs), 
              da.einsum(signature, *da_inputs))
```